src/screens/Departures/state/stop-place-details-state.ts

```diff
@@ -0,0 +1,135 @@
+import {useCallback, useEffect} from 'react';
+import useReducerWithSideEffects, {
+  NoUpdate,
+  ReducerWithSideEffects,
+  Update,
+  UpdateWithSideEffect,
+} from 'use-reducer-with-side-effects';
+import {ErrorType, getAxiosErrorType} from '@atb/api/utils';
+import {getStopsDetails} from '@atb/api/departures/stops-nearest';
+import {StopsDetailsQuery} from '@atb/api/types/generated/StopsDetailsQuery';
+
+export type StopsDetailsDataState = {
+  data: StopsDetailsQuery | null;
+  error?: {type: ErrorType};
+  ids?: Array<string>;
+  isLoading: boolean;
+};
+
+const initialState: StopsDetailsDataState = {
+  data: null,
+  error: undefined,
+  ids: undefined,
+  isLoading: false,
+};
+
+type StopsDetailsDataActions =
+  | {
+      type: 'LOAD_DETAILS';
+      locations?: Array<string>;
+    }
+  | {
+      type: 'STOP_LOADER';
+    }
+  | {
+      type: 'UPDATE_STOP_DETAILS';
+      ids?: Array<string>;
+      result: StopsDetailsQuery;
+    }
+  | {
+      type: 'SET_ERROR';
+      error: ErrorType;
+    };
+
+const reducer: ReducerWithSideEffects<
+  StopsDetailsDataState,
+  StopsDetailsDataActions
+> = (state, action) => {
+  switch (action.type) {
+    case 'LOAD_DETAILS': {
+      if (!action.locations) return NoUpdate();
+      return UpdateWithSideEffect<
+        StopsDetailsDataState,
+        StopsDetailsDataActions
+      >(
+        {
+          ...state,
+          isLoading: true,
+          error: undefined,
+        },
+        async (state, dispatch) => {
+          if (!action.locations) return;
+          try {
+            const result = await getStopsDetails({ids: action.locations});
+            dispatch({
+              type: 'UPDATE_STOP_DETAILS',
+              ids: action.locations,
+              result,
+            });
+          } catch (e) {
+            dispatch({
+              type: 'SET_ERROR',
+              error: getAxiosErrorType(e),
+            });
+          } finally {
+            dispatch({type: 'STOP_LOADER'});
+          }
+        },
+      );
+    }
+
+    case 'STOP_LOADER': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+      });
+    }
+
+    case 'UPDATE_STOP_DETAILS': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+        ids: action.ids,
+        data: action.result,
+      });
+    }
+
+    case 'SET_ERROR': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        error: {
+          type: action.error,
+        },
+        data: state.data,
+      });
+    }
+
+    default:
+      return NoUpdate();
+  }
```

Comment:
Hvis mulig gjerne ta bort `default`-handling her. Vet ikke om det er noe spesielt med `UseReducerWithSideEffect` som gjør at den må være tilstede, men vanligvis er det en risiko for at `default`-handling maskerer ting man har glemt å håndtere.

Reply:
Viktigste er at den returnerer `NoUpdate` fremfor undefined/ingenting f.eks. Det er ikke exhaustive switches uansett, så konsekvensen blir stort sett den samme med eller uten default med mindre man thrower en error. Men da kan man ende med motsatt problem (altså ikke komplett case fører til krasjing, og her føler jeg manglende effekt er bedre enn krasjing)

Reply:
Hvis de ikke er exhaustive så er det bare å se bort fra kommentaren.